api/errors/errors.go

```diff
@@ -0,0 +1,318 @@
+	return newWellKnownYarpcError(codes.Cancelled, keyValues)
+}
+
+// InvalidArgument returns a new yarpc error with code InvalidArgument.
+func InvalidArgument(keyValues ...string) error {
+	return newWellKnownYarpcError(codes.InvalidArgument, keyValues)
+}
+
+// DeadlineExceeded returns a new yarpc error with code DeadlineExceeded.
+func DeadlineExceeded(keyValues ...string) error {
+	return newWellKnownYarpcError(codes.DeadlineExceeded, keyValues)
+}
+
+// NotFound returns a new yarpc error with code NotFound.
+func NotFound(keyValues ...string) error {
+	return newWellKnownYarpcError(codes.NotFound, keyValues)
+}
+
+// AlreadyExists returns a new yarpc error with code AlreadyExists.
+func AlreadyExists(keyValues ...string) error {
+	return newWellKnownYarpcError(codes.AlreadyExists, keyValues)
+}
+
+// PermissionDenied returns a new yarpc error with code PermissionDenied.
+func PermissionDenied(keyValues ...string) error {
+	return newWellKnownYarpcError(codes.PermissionDenied, keyValues)
+}
+
+// ResourceExhausted returns a new yarpc error with code ResourceExhausted.
+func ResourceExhausted(keyValues ...string) error {
+	return newWellKnownYarpcError(codes.ResourceExhausted, keyValues)
+}
+
+// FailedPrecondition returns a new yarpc error with code FailedPrecondition.
+func FailedPrecondition(keyValues ...string) error {
+	return newWellKnownYarpcError(codes.FailedPrecondition, keyValues)
+}
+
+// Aborted returns a new yarpc error with code Aborted.
+func Aborted(keyValues ...string) error {
+	return newWellKnownYarpcError(codes.Aborted, keyValues)
+}
+
+// OutOfRange returns a new yarpc error with code OutOfRange.
+func OutOfRange(keyValues ...string) error {
+	return newWellKnownYarpcError(codes.OutOfRange, keyValues)
+}
+
+// Unimplemented returns a new yarpc error with code Unimplemented.
+func Unimplemented(keyValues ...string) error {
+	return newWellKnownYarpcError(codes.Unimplemented, keyValues)
+}
+
+// Internal returns a new yarpc error with code Internal.
+func Internal(keyValues ...string) error {
+	return newWellKnownYarpcError(codes.Internal, keyValues)
+}
+
+// Unavailable returns a new yarpc error with code Unavailable.
+func Unavailable(keyValues ...string) error {
+	return newWellKnownYarpcError(codes.Unavailable, keyValues)
+}
+
+// DataLoss returns a new yarpc error with code DataLoss.
+func DataLoss(keyValues ...string) error {
+	return newWellKnownYarpcError(codes.DataLoss, keyValues)
+}
+
+// Unauthenticated returns a new yarpc error with code Unauthenticated.
+func Unauthenticated(keyValues ...string) error {
+	return newWellKnownYarpcError(codes.Unauthenticated, keyValues)
+}
+
+// Application returns a new yarpc error with code Application and the given user-defined name.
+//
+// TODO: Validate that the name is only lowercase letters and dashes, and figure out what to do if not.
+func Application(name string, keyValues ...string) error {
+	return newUserDefinedYarpcError(name, keyValues)
+}
```

Comment:
We’ve realized that this constructor can be merged with the UnknownError(name, fmt, ...) function.